app/controllers/api/v1/profiles_controller.rb

```diff
@@ -0,0 +1,28 @@
+module Api
+  module V1
+    class ProfilesController < ApplicationController
+      before_action :doorkeeper_authorize!
+      before_action :collect_users, only: :all
+
+      def me
+        render json: current_resource_owner
+      end
+
+      def all
+        render json: @users_hash
+      end
+
+      private
+
+      def collect_users
+        @users_hash = {}
+        User.all.each { |user| @users_hash["user_#{user.id}".to_sym] = user }
```

Comment:
Hint 1: User.where.not(id: current_resourse_owner.id)
Hint 2: users.collect{ |u| ["user_#{u.id}", user] }.to_h
Hint 3: Combine the two: User.where.not(id: current_resourse_owner.id).collect{ |u| ["user_#{u.id}", user] }.to_h

So this way you will get a one-liner:
```
def collect_users
  User.where.not(id: current_resourse_owner.id).collect{ |u| ["user_#{u.id}", user] }.to_h
end
```

Reply:
Even better:
```
class User
  scope except_user, -> (user_id) { where.not(id: user.id) }
end

def users_except_current
  User.except_user(current_resourse_owner.id))
end

def collect_users
  build_hash(users_except_current)
end

def build_hash(users)
  users.collect{ |u| ["user_#{u.id}", u] }.to_h
end

```